staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -263,9 +264,10 @@ type Cacher struct {
 	// indexedTrigger is used for optimizing amount of watchers that needs to process
 	// an incoming event.
 	indexedTrigger *indexedTriggerFunc
-	// watchers is mapping from the value of trigger function that a
-	// watcher is interested into the watchers
+	
 	watcherIdx int
+	// watchers is mapping from the value of two trigger functions, cacheWatcher and watchersMap,
+	// that a watcher is interested into the watchers
```

Comment:
I'm not sure I understand this.

My understanding of `indexedWatchers` is as follows:
- `watcherIdx` is an increasing index variable used to identify the watchers that watch from the Cache.
- If `indexTriggerFunc` is defined and trigger is supported based on value from a trigger function, we can narrow down the number of watchers that should have to process an event (filtering based on trigger value).
    - If this is the case, then, `indexedWatchers` is a mapping from value of trigger function that a watcher is interested in, to these watchers (technically it is a mapping from both `watchIdx` AND value of trigger function to interested watchers).
- However, if the above is not true, then the mapping exists between `watcherIdx` and the watcher that this index identifier.

Feel free to condense and reword as you see fit :)

Reply:
Thanks, this is done :)